src/structures/MessageAttachment.js

```diff
@@ -43,6 +43,20 @@ class MessageAttachment {
     return this;
   }
 
+  /**
+   * Sets whether this attachment is a spoiler
+   * @param {boolean} [spoiler=true] Whether the attachment should be marked as a spoiler
+   * @returns {MessageAttachment} This attachment
+   */
+  setSpoiler(spoiler = true) {
+    if (spoiler === true && !this.name.startsWith('SPOILER_')) {
+      this.name = `SPOILER_${this.name}`;
+      return this;
+    }
+    if (spoiler === false && this.name.startsWith('SPOILER_')) this.name.replace('SPOILER_', '');
```

Comment:
String.prototype.replace does not change the original string, so this is needed
```suggestion
    if (spoiler === false && this.name.startsWith('SPOILER_')) {
        this.name = this.name.replace('SPOILER_', '');
    }
```

Reply:
Yep, seems like I forgot about that, since commits ago.

Reply:
Wait, couldn't it just be `if (spoiler === false && this.name.startsWith('SPOILER_')) this.name = this.name.replace('SPOILER_', '');`? I recall @ ckohen suggesting to disregard the if/else chain, so...

Reply:
What are you suggesting? I just moved it to a new like because otherwise ESLint would complain

Reply:
Woops